hawkular-alerts-bus/src/main/java/org/hawkular/alerts/bus/listener/ActionPluginOperationListener.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.alerts.bus.listener;
+
+import javax.ejb.ActivationConfigProperty;
+import javax.ejb.EJB;
+import javax.ejb.MessageDriven;
+import javax.ejb.TransactionAttribute;
+import javax.ejb.TransactionAttributeType;
+import javax.jms.MessageListener;
+
+import org.hawkular.alerts.api.json.JsonUtil;
+import org.hawkular.alerts.api.model.action.Action;
+import org.hawkular.alerts.api.services.ActionsService;
+import org.hawkular.alerts.bus.api.BusOperationMessage;
+import org.hawkular.alerts.bus.log.MsgLogger;
+import org.hawkular.bus.common.consumer.BasicMessageListener;
+import org.jboss.logging.Logger;
+
+/**
+ * A component that listens from the bus operation messages coming from plugins.
+ *
+ * @author Jay Shaughnessy
+ * @author Lucas Ponce
+ */
+@MessageDriven(messageListenerInterface = MessageListener.class, activationConfig = {
+        @ActivationConfigProperty(propertyName = "destinationType", propertyValue = "javax.jms.Queue"),
+        @ActivationConfigProperty(propertyName = "destination", propertyValue = "HawkularAlertsOperationsQueue")})
```

Comment:
The main design is that we are going to have a plugin developers.
They have to interact only with the org.hawkular.alerts.actions.api.
In that package the @Plugin defines a plugin implementation, this class must implement ActionPluginListener.process(ActionMessage action), which is the main method that "listens" Actions.

The "glue" code is completely hidden, a plugin developer can write its code without any worry if his code should be a JMS MDB or other implementation, as we have a hawkular deployment and a standalone deployment.

This is intended to hide all bus dependencies on the plugin side, and also to simplify the design.

This MDB is part of the implementation used in hawkular, but the standalone has other with no JMS involved.

Ok, main feature of the plugin is to listen Actions and process them. But now, we offer bidirectional communication, so a plugin can send messages to the engine.

I didn't choose to send an Action, because in essence the plugin should not modify an action, apart of the result, so sending the whole action back I think it was too heavy (indeed at some point we could review where we can lean payloads, but that's another fight for future).

So, before that plugins can send explicitly messages to the engine, we used a first one, called operation, to register a plugin into the engine (without it, the engine has not visibility of plugins deployed in an installation). My first approach was to reuse the concept of OperationMessage as a generic way that can be extended. I thought also in PluginMessage or ResultMessage as an alternatives.

In summary, we have ActionMessages that are coming from engine to the plugins, and OperationMessages that are sending from the plugins to the engine.

Writting this I think that perhaps a PluginMessage might be more meaningful as name.

I can change it later.